src/icc-tags.js

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import {iccTags, iccProfile} from './icc-tag-names';
+
+export default {
+    read
+};
+
+const PROFILE_HEADER_LENGTH = 84;
+
+function read(dataView, iccData) {
+    try {
+        const totalIccProfileLength = iccData.reduce((sum, icc) => sum + icc.length, 0);
+
+        const iccBinaryData = new Uint8Array(totalIccProfileLength);
+        let offset = 0;
+        const buffer = dataView.buffer;
+        for (let chunkNumber = 1; chunkNumber <= iccData.length; chunkNumber++) {
+            const iccDataChunk = iccData.find((x) => x.chunkNumber === chunkNumber);
+            if (!iccDataChunk) {
+                throw new Error(`ICC chunk ${chunkNumber} not found`);
+            }
+
+            const data = buffer.slice(iccDataChunk.offset, iccDataChunk.offset + iccDataChunk.length);
+            const chunkData = new Uint8Array(data);
+
+            iccBinaryData.set(chunkData, offset);
+            offset += chunkData.length;
+        }
+
+        return parseTags(new DataView(iccBinaryData.buffer));
+    } catch (error) {
+        return {};
+    }
+}
+
+const getContentAtOffsetAsString = (offset, buffer) => {
+    return String.fromCharCode.apply(null, new Uint8Array(buffer.slice(offset, offset + 4))).trim();
+};
+
+function sliceToString(slice) {
+    return String.fromCharCode.apply(null, new Uint8Array(slice));
+}
+
+function parseTags(dataView) {
+    const buffer = dataView.buffer;
+
+    const length = dataView.getUint32();
+    if (dataView.length !== length) {
+        new Error('ICC profile length not matching');
+    }
+
+    if (dataView.length < PROFILE_HEADER_LENGTH) {
+        new Error('ICC profile too short');
+    }
+
+    const retVal = {};
+
+    for (const [offset, profileEntry] of Object.entries(iccProfile)) {
+        const value = profileEntry.value(dataView, parseInt(offset));
+        let description = value;
+        if (profileEntry.description) {
+            description = profileEntry.description(value);
+        }
+
+        retVal[profileEntry.name] = {
+            value,
+            description
+        };
+    }
+
+    // const signature = sliceToString(buffer.slice(36, 40));
+    // if (signature !== 'acsp') {
+    //     throw new Error('ICC profile: missing signature');
+    // }
+
+    const tagCount = dataView.getUint32(128);
+    let tagHeaderOffset = 132;
+
+    for (let i = 0; i < tagCount; i++) {
+        const tagSignature = getContentAtOffsetAsString(tagHeaderOffset, buffer);
+        const tagOffset = dataView.getUint32(tagHeaderOffset + 4);
+        const tagSize = dataView.getUint32(tagHeaderOffset + 8);
+        if (tagOffset > buffer.length) {
+            throw new Error('Invalid ICC profile: Tag offset out of bounds');
```

Comment:
Does this really need to abort everything? Would it be possible to still read some of the tags?